server/backend/check.ml

```diff
@@ -139,27 +148,46 @@ if [ $res = 0 ]; then
 fi
 |}
 
+let with_test_windows pkg = {|
+if [ $res = 0 ]; then
+    ocaml-env exec --64 -- opam remove -y "|}^pkg^{|"
+    ocaml-env exec --64 -- opam install -vty "|}^pkg^{|"
+    res=$?
+    if [ $res = 20 ]; then
+        res=0
+    fi
+fi
+|}
+
 let with_test ~conf pkg =
   if Server_configfile.with_test conf then
-    with_test pkg
+    match platform_os conf with `Linux -> with_test_linux pkg | `Windows -> with_test_windows pkg
   else
     ""
 
-let with_lower_bound pkg = {|
+let with_lower_bound_linux pkg = {|
 if [ $res = 0 ]; then
     opam remove -y "|}^pkg^{|"
     env OPAMCRITERIA="+removed,+count[version-lag,solution]" opam install -vy "|}^pkg^{|"
     res=$?
 fi
 |}
 
+let with_lower_bound_windows pkg = {|
+if [ $res = 0 ]; then
+    ocaml-env exec --64 -- opam remove -y "|}^pkg^{|"
+    ocaml-env exec --64 -- env OPAMCRITERIA="+removed,+count[version-lag,solution]" opam install -vy "|}^pkg^{|"
+    res=$?
+fi
+|}
```

Comment:
instead of duplicating every functions I think we should aim at factorising things. For example:
```suggestion
let with_lower_bound ~windows pkg =
  let opam = if windows then "ocaml-env exec --64 -- opam" else "opam" in
  {|
if [ $res = 0 ]; then
  |}^opam^{| remove -y "|}^pkg^{|"
  |}^opam^{| install --criteria="+removed,+count[version-lag,solution]" -vy "|}^pkg^{|"
  res=$?
fi
|}
```

Reply:
That's sensible for these small functions.

Reply:
Is it not possible for the main function?